packages/flutter/test/widgets/two_dimensional_viewport_test.dart

```diff
@@ -0,0 +1,852 @@
+      // Horizontal
+      final ScrollableState horizontal = Scrollable.of(capturedContext, axis: Axis.horizontal);
+      expect(horizontal.widget.controller, isNotNull);
+    }, variant: TargetPlatformVariant.all());
+
+    testWidgets('asserts the axis directions do not conflict with one another', (WidgetTester tester) async {
+      final List<Object> exceptions = <Object>[];
+      final FlutterExceptionHandler? oldHandler = FlutterError.onError;
+      FlutterError.onError = (FlutterErrorDetails details) {
+        exceptions.add(details.exception);
+      };
+      // Horizontal mismatch
+      await tester.pumpWidget(TwoDimensionalScrollable(
+        horizontalDetails: const ScrollableDetails.horizontal(),
+        verticalDetails: const ScrollableDetails.horizontal(),
+        viewportBuilder: (BuildContext context, ViewportOffset verticalPosition, ViewportOffset horizontalPosition) {
+          return Container();
+        },
+      ));
+
+      // Vertical mismatch
+      await tester.pumpWidget(TwoDimensionalScrollable(
+        horizontalDetails: const ScrollableDetails.vertical(),
+        verticalDetails: const ScrollableDetails.vertical(),
+        viewportBuilder: (BuildContext context, ViewportOffset verticalPosition, ViewportOffset horizontalPosition) {
+          return Container();
+        },
+      ));
+
+      // Both
+      await tester.pumpWidget(TwoDimensionalScrollable(
+        horizontalDetails: const ScrollableDetails.vertical(),
+        verticalDetails: const ScrollableDetails.horizontal(),
+        viewportBuilder: (BuildContext context, ViewportOffset verticalPosition, ViewportOffset horizontalPosition) {
+          return Container();
+        },
+      ));
+
+      expect(exceptions.length, 3);
+      for (final Object exception in exceptions) {
+        expect(exception, isAssertionError);
+        expect((exception as AssertionError).message, contains('are not Axis'));
+      }
+      FlutterError.onError = oldHandler;
+    }, variant: TargetPlatformVariant.all());
+
+    testWidgets('correctly sets restorationIds', (WidgetTester tester) async {
+      late BuildContext capturedContext;
+      // with restorationID set
+      await tester.pumpWidget(TwoDimensionalScrollable(
+        restorationId: 'Custom Restoration ID',
+        horizontalDetails: const ScrollableDetails.horizontal(),
+        verticalDetails: const ScrollableDetails.vertical(),
+        viewportBuilder: (BuildContext context, ViewportOffset verticalPosition, ViewportOffset horizontalPosition) {
+          return SizedBox.square(
+            dimension: 200,
+            child: Builder(
+              builder: (BuildContext context) {
+                capturedContext = context;
+                return Container();
+              },
+            )
+          );
+        },
+      ));
+      await tester.pumpAndSettle();
+
+      // TODO(Piinks): Ask goderbauer why this isn't finding the restoration scope.
```

Comment:
Here is where I wanted to ask you, why cannot I not find the RestorationScope added by the TwoDimensionalScrollable? I tried a couple of different ways, but this expectation always fails.

Reply:
Oh, ^ @goderbauer  to help find it in all this. :)

Reply:
To truly test state restoration, you probably want to utilize the https://master-api.flutter.dev/flutter/flutter_test/WidgetTester/restartAndRestore.html method. Looks for some other tests that use it as example.

Reply:
Oh well I feel silly. Thanks for the pointers!